### PR TITLE
fix spelling errors in doc comments

### PR DESCRIPTION
**Pull request by robjtede**
Run over docs with a code spell checker.

Left in the relevant `.vscode` file with allow list but can remove if anyone has strong opinions.

Future fixes:
`ConnectError::Unresolverd` => `ConnectError::Unresolved` (breaking)

### CODE REVIEW

File: src/guard.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ pub fn Any<F: Guard + 'static>(guard: F) -> AnyGuard {
     AnyGuard(vec![Box::new(guard)])
 }
 
-/// Matches if any of supplied guards matche.
+/// Matches if any of supplied guards match.
```
Comment (JohnTitor) [inline src/guard.rs]: I think it's clearer:

```suggestion
/// Matches any of supplied guards.
```

**Review by JohnTitor — COMMENTED** [review]
Overall looks good! I'm reluctant to introduce `.vscode/settings.json` here, do we really need this?

**Comment by robjtede** [comment]
> do we really need this?

Need? No. I understand the reluctance to introduce editor specific configs.

**Review by JohnTitor — APPROVED** [review]
Thanks!

**Pull request merged**